controllers/admin-api-controller.js

```diff
@@ -0,0 +1,242 @@
+            page: pageTemplate(page)
+          })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.patchPage = async (req, res, next) => {
+    try{
+        const pageId = req.params.pageId
+    
+        const updatedData = req.body
+        const page = await Page.findByIdAndUpdate(pageId, updatedData, {new:true, upsert:true})
+        res.json({ 
+            success: true,
+            page: pageTemplate(page)
+        })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.deletePage = async (req, res, next) => {
+    try{
+        const pageId = req.params.pageId
+    
+        await Page.findByIdAndDelete(pageId)
+        res.json({ success: true })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+// Categories 
+exports.postCategory = async (req, res, next) => {
+    try{
+        const title = req.body.title
+        const handle = req.body.handle
+        const description = req.body.description
+        const products = req.body.products
+        const listed = req.body.listed
+        const featured = req.body.featured
+      
+        let category = new Category({
+            title,
+            handle,
+            description,
+            products,
+            listed,
+            featured
+        })
+        category = await category.save()
+        res.json({ 
+            success: true,
+            category: categoryTemplate(category)
+          })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.patchCategory = async (req, res, next) => {
+    try{
+        const categoryId = req.params.categoryId
+    
+        const category = await Category.findByIdAndUpdate(categoryId, req.body, {new:true, upsert:true})
+        res.json({ 
+            success: true,
+            category: categoryTemplate(category)
+        })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.deleteCategory = async (req, res, next) => {
+    try{
+        const categoryId = req.params.categoryId
+    
+        await Category.findByIdAndDelete(categoryId)
+        res.json({ success: true })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+// Product 
+exports.postProduct = async (req, res, next) => {
+    try{
+        const title = req.body.title
+        const description = req.body.description
+        const price = req.body.price
+        const compareAt = req.body.compareAt
+        const imagePaths = req.body.imagePaths
+        const categories = req.body.categories
+        const shippingCost = req.body.shippingCost
+        const stock = req.body.stock
+        const publish = req.body.publish
+      
+        let product = new Product({
+            title,
+            description,
+            price,
+            compareAt,
+            imagePaths,
+            categories,
+            shippingCost,
+            stock,
+            publish
```

Comment:
I personally prefer, when there is no ambiguity and the value is only used once, not defining a new variable and using directly req.body.shippingCost instead